diff --git a/gio/gdbusconnection.c b/gio/gdbusconnection.c
index a543df9..6922567 100644
--- a/gio/gdbusconnection.c
+++ b/gio/gdbusconnection.c
@@ -4477,10 +4477,7 @@ purge_all_signal_subscriptions (GDBusConnection *connection)
 static GDBusInterfaceVTable *
 _g_dbus_interface_vtable_copy (const GDBusInterfaceVTable *vtable)
 {
-  /* Don't waste memory by copying padding - remember to update this
-   * when changing struct _GDBusInterfaceVTable in gdbusconnection.h
-   */
-  return g_memdup2 ((gconstpointer) vtable, 3 * sizeof (gpointer));
+  return g_memdup2 ((gconstpointer) vtable, sizeof (GDBusInterfaceVTable));
 }
 
 static void
@@ -4494,10 +4491,7 @@ _g_dbus_interface_vtable_free (GDBusInterfaceVTable *vtable)
 static GDBusSubtreeVTable *
 _g_dbus_subtree_vtable_copy (const GDBusSubtreeVTable *vtable)
 {
-  /* Don't waste memory by copying padding - remember to update this
-   * when changing struct _GDBusSubtreeVTable in gdbusconnection.h
-   */
-  return g_memdup2 ((gconstpointer) vtable, 3 * sizeof (gpointer));
+  return g_memdup2 ((gconstpointer) vtable, sizeof (GDBusSubtreeVTable));
 }
 
 static void
diff --git a/gio/gdbusconnection.h b/gio/gdbusconnection.h
index 3a8cb9d..395348e 100644
--- a/gio/gdbusconnection.h
+++ b/gio/gdbusconnection.h
@@ -395,10 +395,7 @@ struct _GDBusInterfaceVTable
   GDBusInterfaceSetPropertyFunc set_property;
 
   /*< private >*/
-  /* Padding for future expansion - also remember to update
-   * gdbusconnection.c:_g_dbus_interface_vtable_copy() when
-   * changing this.
-   */
+  /* Padding for future expansion */
   gpointer padding[8];
 };
 
@@ -533,10 +530,7 @@ struct _GDBusSubtreeVTable
   GDBusSubtreeDispatchFunc   dispatch;
 
   /*< private >*/
-  /* Padding for future expansion - also remember to update
-   * gdbusconnection.c:_g_dbus_subtree_vtable_copy() when
-   * changing this.
-   */
+  /* Padding for future expansion */
   gpointer padding[8];
 };
 
diff --git a/glib/gvariant.c b/glib/gvariant.c
index ea99e1f..239b1d2 100644
--- a/glib/gvariant.c
+++ b/glib/gvariant.c
@@ -2986,7 +2986,7 @@ g_variant_iter_new (GVariant *value)
 {
   GVariantIter *iter;
 
-  iter = (GVariantIter *) g_slice_new (struct heap_iter);
+  iter = g_slice_new (GVariantIter);
   GVHI(iter)->value_ref = g_variant_ref (value);
   GVHI(iter)->magic = GVHI_MAGIC;
 
