Index: glib2.0-2.80.1/gio/glocalfile.c
===================================================================
--- glib2.0-2.80.1.orig/gio/glocalfile.c
+++ glib2.0-2.80.1/gio/glocalfile.c
@@ -2253,6 +2253,7 @@ g_local_file_trash (GFile         *file,
               if (basename_len <= strlen (".trashinfo"))
                 break; /* fail with ENAMETOOLONG */
               basename_len -= strlen (".trashinfo");
+              memmove (basename, basename + strlen (".trashinfo"), basename_len);
               basename[basename_len] = '\0';
               i = 1;
               continue;
@@ -2276,6 +2277,7 @@ g_local_file_trash (GFile         *file,
               if (basename_len <= strlen (".XXXXXX"))
                 break; /* fail with ENAMETOOLONG */
               basename_len -= strlen (".XXXXXX");
+              memmove (basename, basename + strlen (".XXXXXX"), basename_len);
               basename[basename_len] = '\0';
               i = 1;
               g_clear_error (&my_error);
